### PR TITLE
[fix scitedotai/scite#3194] Add citation search to OS context menu

### PR DESCRIPTION
**Pull request by bananaoomarang**
+ Add citation search link to OS context menu
+ Update React to 17 + bump other non breaking deps

I tested in Chrome and FF but need some mac man to test on Safari and I can test on Edge I guess once the package is rolled out.

### CODE REVIEW

File: src/background.js [hunk 1]
```diff
@@ -10,3 +10,27 @@ browser.runtime.onInstalled.addListener(function (details) {
...
+  id: 'scite-citation-search',
+  title: 'Search citation statements',
+  contexts: ['selection']
+}, onCreated)
+
+browser.contextMenus.onClicked.addListener(function (info, tab) {
+  if (info.menuItemId === 'scite-citation-search') {
+    if (info.selectionText) {
+      browser.tabs.create({
+        url: `https://scite.ai/search/citations?q=${info.selectionText}`
```
Comment (u-ashish) [inline src/background.js:32]: can we utm this too 😄

Comment (domenicrosati) [reply src/background.js:32]:
```suggestion
        url: `https://scite.ai/search/citations?q=${info.selectionText}&utm_source=generic&utm_medium=plugin&utm_campaign=plugin-citation-search`
```

Comment (domenicrosati) [inline src/background.js]: @bananaoomarang  I think you will want to URI encode this text otherwise a user might break it by selecting illegal synatx

File: src/background.js [hunk 2]
```diff
@@ -10,3 +10,27 @@ browser.runtime.onInstalled.addListener(function (details) {
...
+  id: 'scite-citation-search',
+  title: 'Search scite.ai citation statements',
+  contexts: ['selection']
+}, onCreated)
+
+browser.contextMenus.onClicked.addListener(function (info, tab) {
+  if (info.menuItemId === 'scite-citation-search') {
+    if (info.selectionText) {
+      browser.tabs.create({
+        url: `https://scite.ai/search/citations?q=${info.selectionText}&utm_source=generic&utm_medium=plugin&utm_campaign=plugin-citation-search`
```
Comment (domenicrosati) [inline src/background.js:32]: @bananaoomarang I think you will want to URI encode this text otherwise a user could select illegal text I think

Comment (domenicrosati) [reply src/background.js:32]: weiird chrome automatically encodes it and firefox doesnt

Comment (bananaoomarang) [reply src/background.js]: Just updated for this I think

**Review by domenicrosati — APPROVED** [review]
I can add to Safari
